src/app/Library/CrudPanel/Traits/Update.php

```diff
@@ -17,25 +17,25 @@ trait Update
      * Update a row in the database.
      *
      * @param  int  $id  The entity's id
-     * @param  array  $data  All inputs to be updated.
+     * @param  array  $input  All inputs to be updated.
      * @return object
      */
-    public function update($id, $data)
+    public function update($id, $input)
     {
-        $data = $this->decodeJsonCastedAttributes($data);
-        $data = $this->compactFakeFields($data);
+        $input = $this->decodeJsonCastedAttributes($input);
+        $input = $this->compactFakeFields($input);
         $item = $this->model->findOrFail($id);
 
-        $data = $this->changeBelongsToNamesFromRelationshipToForeignKey($data);
+        $input = $this->changeBelongsToNamesFromRelationshipToForeignKey($input);
 
-        $this->createRelations($item, $data);
+        $relation_input = $this->getRelationDetailsFromInput($input);
 
-        // omit the n-n relationships when updating the eloquent item
-        $nn_relationships = Arr::pluck($this->getRelationFieldsWithPivot(), 'name');
+        // handle the creation of the model relations.
+        $this->createRelationsForItem($item, $relation_input);
 
-        $data = Arr::except($data, $nn_relationships);
+        $field_names_to_exclude = $this->getFieldNamesToExcludeFromInput($this->getRelationFieldsWithoutRelationType('BelongsTo', true));
```

Comment:
```suggestion
        $field_names_to_exclude = $this->getFieldNamesBeforeFirstDot($this->getRelationFieldsWithoutRelationType('BelongsTo', true));
```

Reply:
See?

- variable name tells what you are going to do with it (exclude)
- each function name tells what that function receives/does/returns

Much more clear, right?